ui/src/index.tsx

```diff
@@ -12,21 +12,37 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-import { initializeIcons, loadTheme } from '@fluentui/react'
+import { loadTheme, registerIcons } from '@fluentui/react'
 import React from 'react'
 import ReactDOM from 'react-dom'
 import './cards'
 import './index.scss'
 import Router from './router'
 import * as serviceWorker from './serviceWorker'
 import { defaultTheme } from './theme'
+import * as Icons from '@fluentui/react-icons-mdl2'
+
 
 loadTheme({
   defaultFontStyle: { fontFamily: 'Inter' },
   palette: defaultTheme.fluentPalette,
 })
 
-initializeIcons('fonts/icons/')
+type IconMapping = keyof typeof Icons
+
+const reactIcons = Object.keys(Icons).filter((icon) => {
+  return Icons[icon as IconMapping].name === 'Component'
+})
+
+const iconsToRegister = (reactIcons as IconMapping[]).reduce(
+  (acc: { [key: string]: React.FunctionComponentElement<React.FC> }, cur) => {
+    acc[cur.substring(0, cur.length - 4)] = React.createElement<React.FC>(Icons[cur] as React.FC)
+    return acc
+  }, {})
+
+registerIcons({
+  icons: iconsToRegister
+})
```

Comment:
This should be cleaner. Can you verify this works as expected @marek-mihok? Also note that `icons` prop in `registerIcons` accepts `string | JSX.Element` so that your original typing was a bit misleading (using React.FC instead).
 
```suggestion
const icons = Object.entries(Icons).reduce((acc, [iconName, iconComponent]) => {
  if (iconComponent.name === 'Component') acc[iconName.slice(0, -4)] = React.createElement(iconComponent as React.FC)
  return acc
}, {} as { [key: string]: JSX.Element })
registerIcons({ icons })
```